### PR TITLE
Document decimal type mapping in MongoDB connector

### LINKED ISSUE #17322: Update documentation for type mapping in MongoDB connector
https://trino.io/docs/current/connector/mongodb.html?highlight=mongodb#type-mapping - Docs doesn't specify the support for Decimal data type in MongoDB connector but the [test coverage](https://github.com/trinodb/trino/blob/24c0f6272f2ff285831a0251d02afc9b7d1b6cf1/plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoTypeMapping.java#L164) proves we support decimal type. So we might need to make the doc in sync for MongoDB connector

### PR DESCRIPTION
**Pull request by mosabua**
## Description

- For the MongoDB connector

## Additional context and related issues

Fixes #17322

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: docs/src/main/sphinx/connector/mongodb.rst [hunk 1]
```diff
@@ -404,6 +404,9 @@ this table:
   * - ``Double``
     - ``DOUBLE``
     -
+  * - ``Decimal(p, s)``
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/mongodb.rst:407]:
```suggestion
  * - ``Decimal128``
```

File: docs/src/main/sphinx/connector/mongodb.rst [hunk 2]
```diff
@@ -446,6 +449,8 @@ this table:
     - ``Int64``
   * - ``DOUBLE``
     - ``Double``
+  * - ``DECIMAL(p, s)``
+    - ``Decimal(p, s)``
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/mongodb.rst:453]:
```suggestion
    - ``Decimal128``
```

**Review by ebyhr — APPROVED** [review]
I would include the connector name in the commit title instead of the commit body.

**Comment by mosabua** [comment]
Related question @ebyhr  .. the mapping has a ObjectId data type .. but from what I know there is no such thing in Trino .. is it just a random string .. or an int or UUID or something else ..  once I know I can send another PR .. just wont have time to dig in the code.

**Comment by ebyhr** [comment]
The type is implemented within MongoDB connector. 
https://github.com/trinodb/trino/blob/master/plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/ObjectIdType.java

**Pull request merged**